script.js

```diff
@@ -1,70 +1,96 @@
 
-      TOMApp.prototype.updateproducthtml = function() {
-        for( i=0; i< this.products.length ; i++){
-          this.products[i].updatehtml();
+      TOMApp.prototype.initProducts = function() {
+        var self = this;
+
+        for (var i=0; i<productData.sales.length ; i++) {
+          self.products.push(new Product(productData.sales[i], i)  );
         }
       }
 
-      TOMApp.prototype.updatedom = function() {
-        thishtml=''; 
-        for(var i=0; i< this.products.length ; i++){
+      TOMApp.prototype.render = function() {
+        var self = this;
+
+        self.regions.$mainRegion.empty();
+
+        var $row = null;
+        for(var i=0; i< self.products.length ; i++){
           
           if (i % 3 == 0 ) {  
-            thishtml += "<div class='row'>"; 
+            $row = $("<div>").addClass("row");
             console.log("START");
           }
 
-          thishtml += this.products[i].htmlview;
-          if ((i % 3 == 2) || i == (this.products.length-1) ) {
-            thishtml += "</div>";
+          $row.append(self.products[i].render());
+
+          if ((i % 3 == 2) || i == (self.products.length-1)) {
+            self.regions.$mainRegion.append($row);
             console.log("FINISH");
           }
         }
-        $("#content").append(thishtml)
       }
 
-      function productobj(product, i) {
-        this.photo        = product.photos.medium_half;
-        this.title        = product.name;
-        this.tagline      = product.tagline;
-        this.url          = product.url;
-        this.htmlview     = "";
-        this.index        = i;
-        this.custom_class = "col"+ ((i % 3) +1);
+      function Product(product, i) {
+        var self = this;
+
+        self.photo        = product.photos.medium_half;
+        self.title        = product.name;
+        self.tagline      = product.tagline;
+        self.url          = product.url;
+        self.index        = i;
+        self.custom_class = "col"+ ((i % 3) +1);
+
+        self.$elem = null;
       }
 
-      productobj.prototype.updatehtml = function() {
+      Product.prototype.render = function() {
         var self = this;
-        self.htmlview = 
-          template.replace('{image}', self.photo)
-                  .replace('{title}', self.title)
-                  .replace('{tagline}', self.tagline)
-                  .replace('{url}', self.url)
-                  .replace('{custom_class}', self.custom_class);
+
+        var param = {
+          image: self.photo,
+          title: self.title,
+          tagline: self.tagline,
+          url: self.url,
+          custom_class: self.custom_class
+        }
+
+        var htmlView = _template(productTemplateHtml, param);
+
+        if (self.$elem) {
+          // instantiated already, just update html
+          self.$elem.html(htmlView);
+        } else {
+          // Instantiation of the jQuery object
+          self.$elem = $(htmlView);
+        }
+
+        return self.$elem;
```

Comment:
We are now creating and returning jQuery DOMs instead of simply interpolating/rendering the html.